LiteCore/Storage/SQLiteKeyStore.cc

```diff
@@ -97,7 +95,7 @@ namespace litecore {
 
 
     void SQLiteKeyStore::reopen() {
-        if (_existence == kNonexistent)
+        if (!db().keyStoreExists(_name))
```

Comment:
I meant the above if-test, whether it is necessary?

Reply:
I know what you mean, and my answer was based on that.  Once the createTable method is entered there is no way to determine how to correctly set the _uncommittedTable boolean.  The only way is to check if it is already existing.

Reply:
I see, you wanted to be sure that the following create table actually creates one.

Reply:
Reasonably sure, yes.  Since it is a check and create, there is still room for racing but the penalty for losing the race is small (losing cached statements)